sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -982,6 +982,50 @@ ssize_t coap_opt_finish(coap_pkt_t *pkt, uint16_t flags)
     return pkt->payload - (uint8_t *)pkt->hdr;
 }
 
+ssize_t coap_opt_remove(coap_pkt_t *pkt, uint16_t opt_num)
+{
+    coap_optpos_t *optpos = pkt->options;
+    coap_optpos_t *prev_opt = NULL;
+    unsigned opt_count = pkt->options_len;
+    uint8_t *pkt_start = (uint8_t *)pkt->hdr;
+    size_t removed = 0;
+
+    while (opt_count--) {
+        if ((removed == 0) && (optpos->opt_num == opt_num)) {
+            prev_opt = optpos;
+            optpos++;
+            removed = optpos->offset - prev_opt->offset;
+        }
+        if ((removed > 0) && (optpos->opt_num >= opt_num)) {
```

Comment:
The second half of that statement is automatically true due to the options being sorted.

If you want to still check that, maybe:

```suggestion
        if (removed > 0) {
            assert(optpos->opt_num >= opt_num);
```

but just plain `if >` would work too IMO.